Jetpack Backup: scroll back to credentials settings when expanding credentials

Initsogar opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Resolves https://github.com/Automattic/jetpack-backup-team/issues/449

In Calypso Blue, if we are trying to edit server credentials, it goes to the top of the page. The idea is to prevent scrolling to the top. Here is a quick demo of current behavior:

https://github.com/Automattic/wp-calypso/assets/1488641/277cd659-92af-4917-a71f-d893074b79d1

## Proposed Changes
* Remove `setScroll` callback from Jetpack settings in Calypso Blue router definition. This will prevent Jetpack settings to scrolling up to the top when navigating on the credentials settings.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Spin up a Calypso live branch
* Pick a site with Jetpack VaultPress Backup and credentials configured
* Navigate to Settings > Jetpack (or `/settings/jetpack/site_slug`)
* Resize the window to make it a bit smaller
* Scroll down to `Remote server credentials settings`. It should look like this:
  ![CleanShot 2024-04-23 at 17 05 20@2x](https://github.com/Automattic/wp-calypso/assets/1488641/523f099e-78e6-4c30-827b-1164188cb19b)
* Click on the arrow to expand server credentials
* Ensure the browser does not scroll back to the top. It should behave similar to this video:

https://github.com/Automattic/wp-calypso/assets/1488641/5732b2df-04b9-4c4d-b37c-cee8486a0ad7

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Initsogar committed:
Scroll to advanced credentials settings when editing

mavegaf reviewed (approved):
✅  It resolves the issue while editing the credentials. 

Just pointing out that the same issue happens if we want to add new credentials. 
![add_credentials](https://github.com/Automattic/wp-calypso/assets/2747834/650cb1b5-e5f3-41f4-894e-517ff191a2be)

Initsogar committed:
Revert previous changes as they are no longer required

Initsogar committed:
Remove setScroll callback from Jetpack settings in Calypso Blue

This will prevent Jetpack settings to scroll up to the top when navigating on the credentials settings.

Initsogar committed:
Remove unnecessary line break

Initsogar commented:
> ✅ It resolves the issue while editing the credentials.
> Just pointing out that the same issue happens if we want to add new credentials.

Thanks for your feedback @mavegaf! Given that, I pivoted the solution into fixing the root cause, instead of just doing a workaround. I noted that [we are setting the scroll to the top anytime we navigate to Jetpack settings in Calypso Blue](https://github.com/Automattic/wp-calypso/blob/d4b4d4085a4853f0731d7dbd82857813a6b9e0e4/client/my-sites/site-settings/settings-controller.js#L32), so decided to [remove that callback from the router](https://github.com/Automattic/wp-calypso/blob/d4b4d4085a4853f0731d7dbd82857813a6b9e0e4/client/my-sites/site-settings/settings-jetpack/index.js#L26). This only impacts `/settings/jetpack/:site_id` route, which is Jetpack settings page and only Calypso Blue.

Could you take another look?
